core/block_validator.go

```diff
@@ -215,6 +218,9 @@ func (v *BlockValidator) ValidateCXReceiptsProof(cxp *types.CXReceiptsProof) err
 
 	// (2) verify the outgoingCXReceiptsHash match
 	outgoingHashFromSourceShard := crypto.Keccak256Hash(byteBuffer.Bytes())
+	if byteBuffer.Len() == 0 {
+		outgoingHashFromSourceShard = types.EmptyRootHash
+	}
```

Comment:
This should be an no-op as if there is no receipts, then the dest shard shouldn't receive it. But just to make sure the logic are the same with the source side. add it here too.